<launch>
    <!-- weight file path -->
    <arg name="weights" default="$(find yolov8_ros)/scripts/weights/azure_rcjp2022_500_v8n/best.pt"/>    <!-- change the weight path -->

    <!-- probability -->
    <arg name="conf" default="0.3"/>

    <!-- replace imgsz -->
    <arg name="inference_size_h" default="640"/>
    <arg name="inference_size_w" default="480"/>

    <!-- Visualize using OpenCV window -->
    <arg name="view_image" default="false"/>

    <!-- Save result images -->
    <arg name="save_image" default="false"/>

    <!-- Publish Rate -->
    <arg name="pub_rate" default="5"/>

    <!-- image topic name-->
    <arg name="image_topic_name" default="/rgb/image_raw"/>    <!-- change the each camera -->

    <node pkg="yolov8_ros" name="yolov8" type="detect_ros.py" output="screen">
        <param name="weights" value="$(arg weights)"/>
        <param name="conf" value="$(arg conf)"/>

        <param name="inference_size_h" value="$(arg inference_size_h)"/>
        <param name="inference_size_w" value="$(arg inference_size_w)"/>

        <param name="view_image" value="$(arg view_image)"/>
        <param name="save_image" value="$(arg save_image)"/>
        <param name="pub_rate"   value="$(arg pub_rate)"/>

        <param name="initial_predict" type="bool" value="true"/>
        <param name="image_topic_name" value="$(arg image_topic_name)"/>
    </node>


    <!-- point cloud name -->
    <arg name="point_cloud_name" default="/points2"/>    <!-- change the each camera -->

    <include file="$(find bbox_to_tf)/launch/bbox_to_tf.launch">
        <arg name="node_name"                 value="yolov8_bbox_to_tf"/>
        <arg name="base_frame_name"           value="base_footprint"/>
        <arg name="bbox_topic_name"           value="/yolov8/objects_rect"/>
        <arg name="cloud_topic_name"          value="$(arg point_cloud_name)"/>
        <arg name="img_topic_name"            value="$(arg image_topic_name)"/>
        <arg name="execute_default"           value="true"/>
        <arg name="cluster_tolerance"         value="0.008"/>
        <arg name="min_clusterSize"           value="10"/>
        <arg name="max_lusterSize"            value="2000000"/>
        <arg name="noise_point_cloud_range"   value="0.03"/>
        <arg name="rviz"                      value="true"/>
    </include>
</launch>